src/sentry/integrations/slack/utils.py

```diff
@@ -102,7 +102,10 @@ def build_assigned_text(identity, assignee):
         assignee_text = assigned_actor.slug
     elif actor.type == User:
         try:
-            assignee_ident = Identity.objects.get(user=assigned_actor)
+            assignee_ident = Identity.objects.get(
+                user=assigned_actor,
+                idp__organization=group.organization,
```

Comment:
Use `idp__organization_id=group.organization_id` instead. It's not obvious if `group.organization` is guarnateed to be resolved at this point or not.

Reply:
Lastly, what's the guarantee here that this will only return 1 row? The only UNIQUE on this table is for `(idp_id, external_id)`. Otherwise, I feel like it'd still be possible to return more than 1 row, at least technically nothing would prevent it.

Reply:
1144377 For your first comment.

Yeah that's a good point, we probably don't want one user to have multiple `external_ids` for one identity provider. Added the unique constraint here 9ddabae